components/GeoApi.js

```diff
@@ -0,0 +1,84 @@
+import React, { useEffect, useState } from "react";
+import style from "../styles/Home.module.css";
+
+function GeoApi(props) {
+  const { setWeatherForecast, setWeatherNow } = props;
+  const [locationName, setLocationName] = useState("Linn, Kansas");
+  const [searchTerm, setSearchTerm] = useState("");
+  const API_KEY = process.env.NEXT_PUBLIC_GEOLOCATION_API_KEY;
+
+  const handleChange = (e) => {
+    setSearchTerm(e.target.value);
+  };
+
+  //Just keeping this function to help with development, will delete later
+  async function fetchCoords() {
+    if (API_KEY === null) {
+      console.log("Geolocation API key not set.");
+    }
+    let firstRes = await fetch(`http://api.positionstack.com/v1/forward?access_key=${API_KEY}&query=${searchTerm}`);
+    let firstData = await firstRes.json();
+    let a = 0;
+    while (firstData.data[a].continent !== "North America") {
+      a++;
+    }
+    const long = firstData.data[a].longitude;
+    const lat = firstData.data[a].latitude;
+    console.log(a, firstData);
+    return { long, lat };
+  }
+
+  //Gets weather data when Search button clicked
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+
+    //Step 1 -  Fetches the longitude and latitude
+    let long = 0;
+    let lat = 0;
+    let indexA = 0;
+
+    const firstRes = await fetch(`http://api.positionstack.com/v1/forward?access_key=${API_KEY}&query=${searchTerm}`);
```

Comment:
Way to figure out the geocoding with this API!

Reply:
I changed the name of variable `a `to `searchHitIndex`.